rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1032,78 +1032,133 @@ private void processReleaseRequests() {
         );
 
-        releaseRequestQueue.process(MAX_RELEASE_ITERATIONS, (ReleaseRequestQueue.Entry releaseRequest) -> {
-            Optional<ReleaseTransactionBuilder.BuildResult> result = txBuilder.buildAmountTo(
-                    releaseRequest.getDestination(),
-                    releaseRequest.getAmount()
-            );
+        // Instead of creating a pegout transaction for every element in the release request queue,
+        // check if the elapsed time or blocks have gone by before creating the pegout transaction including all elements in the release request queue.
+        // Under a rush hour of peg-outs, the Bridge may need to create more than one peg-out transaction
+        // simultaneously per peg-out event to reduce the transaction size, and input count.
+        // Pending: Define the limit of the transaction size
 
-            // Couldn't build a transaction to release these funds
-            // Log the event and return false so that the request remains in the
-            // queue for future processing.
-            // Further logging is done at the tx builder level.
-            if (!result.isPresent()) {
-                logger.warn(
-                        "Couldn't build a release BTC tx for <{}, {}>",
-                        releaseRequest.getDestination().toBase58(),
-                        releaseRequest.getAmount());
-                return false;
-            }
+        if (activations.isActive(RSKIP271)) {
+            long currentBlockNumber = rskExecutionBlock.getNumber();
+            long nextPegoutCreationBlockNumber = getNextPegoutCreationBlockNumber();
+
+            if (currentBlockNumber >= nextPegoutCreationBlockNumber) {
+                // batch pegout transactions
+                Optional<ReleaseTransactionBuilder.BuildResult> result = txBuilder.buildBatchedPegouts(releaseRequestQueue.getEntries());
+
+                if (!result.isPresent()) {
+                    logger.warn(
+                            "Couldn't build a release BTC tx for <{}, with sum {}>",
+                            releaseRequestQueue.getEntries().hashCode(),
+                            releaseRequestQueue.getEntries().stream().mapToDouble(e -> e.getAmount().value).sum());
+                    return;
+                }
 
-            // We have a BTC transaction, mark the UTXOs as spent and add the tx
-            // to the release set.
+                List<UTXO> availableUTXOs;
+                ReleaseTransactionSet releaseTransactionSet;
+                try {
+                    availableUTXOs = getActiveFederationBtcUTXOs();
+                    releaseTransactionSet = provider.getReleaseTransactionSet();
+                } catch (IOException exception) {
+                    // Unexpected error accessing storage, log and fail
+                    logger.error("Unexpected error accessing storage while attempting to add a release BTC tx", exception);
+                    return;
+                }
```

Comment:
We could move this to before the `if (activations.isActive(RSKIP271)) {` part. That way we are not repeating this code

Reply:
Done